rpc/core/blocks.go

```diff
@@ -24,24 +24,24 @@ func BlockchainInfo(ctx *rpctypes.Context, minHeight, maxHeight int64) (*ctypes.
 	const limit int64 = 20
 	var err error
 	minHeight, maxHeight, err = filterMinMax(
-		env.BlockStore.Base(),
-		env.BlockStore.Height(),
+		GetEnvironment().BlockStore.Base(),
+		GetEnvironment().BlockStore.Height(),
 		minHeight,
 		maxHeight,
 		limit)
 	if err != nil {
 		return nil, err
 	}
-	env.Logger.Debug("BlockchainInfoHandler", "maxHeight", maxHeight, "minHeight", minHeight)
+	GetEnvironment().Logger.Debug("BlockchainInfoHandler", "maxHeight", maxHeight, "minHeight", minHeight)
 
 	blockMetas := []*types.BlockMeta{}
 	for height := maxHeight; height >= minHeight; height-- {
-		blockMeta := env.BlockStore.LoadBlockMeta(height)
+		blockMeta := GetEnvironment().BlockStore.LoadBlockMeta(height)
 		blockMetas = append(blockMetas, blockMeta)
 	}
 
 	return &ctypes.ResultBlockchainInfo{
-		LastHeight: env.BlockStore.Height(),
+		LastHeight: GetEnvironment().BlockStore.Height(),
```

Comment:
I recommend calling GetEnv once to avoid relocking back and forth on each call. It's also correct to do so because the shared memory can potentially change during the execution of such func. In the selected code block, `LoadBlockMeta` can be called from one env, while further `Height` call from another env causing  unpredictable outcome.

Reply:
resolved by https://github.com/celestiaorg/celestia-core/pull/869/commits/fda227f312905d9c4e37b60f9608428ecdfe111a